tests/schema_registry/test_config.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pytest
+
+from confluent_kafka.schema_registry import SchemaRegistryClient
+
+TEST_URL = 'http://SchemaRegistry:65534'
+TEST_USERNAME = 'sr_user'
+TEST_USER_PASSWORD = 'sr_user_secret'
+
+"""
+Tests to ensure all configurations are handled correctly.
+
+"""
+
+
+def test_config_url_invalid():
+    conf = {'url': 'htt://SchemaRegistry:65534'}
+    with pytest.raises(ValueError) as e:
+        SchemaRegistryClient(conf)
+    assert e.match('Invalid url htt://SchemaRegistry:65534')
+
+
+def test_config_url_invalid_type():
+    conf = {'url': dict()}
+    with pytest.raises(ValueError, match="url must be an instance of str not <(.*)>$"):
+        SchemaRegistryClient(conf)
+
+
+def test_config_url_None():
+    conf = {}
+    with pytest.raises(ValueError, match="url must be an instance of str not"
+                                         " <type 'NoneType'>$"):
```

Comment:
That error message isnt great to be honest.
"Missing required configuration property url"